docs/appnotes/0016-authorisation-in-tams-workflows.md

```diff
@@ -0,0 +1,121 @@
+
+Then roles could be created in the authorisation system which allow some combinations of those scopes, for example:
+
+- `administrator`: Has all three scopes
+- `viewer`: Has `tams-api/read`
+- `editor`: Has `tams-api/read` and `tams-api/write`
+- `store-writer`: Has `tams-api/write`
+- `store-cleanup-system`: Has `tams-api/delete`
+
+Users, or groups of users can then be assigned into those roles: for example a "News Journalists" group might be assigned `editor`, while other staff have `viewer`, but only the automated processes of a MAM have `store-cleanup-system`.
+Alternatively, the permissions can be used directly, assigning them to user groups without grouping them into roles.
+
+To implement the authorisation, the authorisation server checks the requested scopes against the user's access when issuing a token, and rejects the request if a suitable access isn't assigned.
+
+## Finer Grained Authorisation
+
+A further build on the very coarse role-based approach above is to expand the set of permissions to apply to specific Sources and Flows.
+However the implementation of this can become complex and unwieldy, especially if each Source and Flow in the system has a separate set of permissions to manage and it becomes necessary to edit them all to implement a policy change.
+
+Attribute-based Access Control (ABAC) is one approach to manage this complexity, by describing permissions policies based on the attributes of resources (Sources and Flows), and if necessary, users as well.
+However full ABAC can be challenging to implement and requires a degree of organisational maturity to construct and manage stable attributes.
+This topic is well outside the scope of this document, however there exists plenty of literature and tools implementing ABAC in general terms.
+
+In practical TAMS solutions, this could look like defining a "class" attribute which can be applied to a Source or Flow, where "class" could also be thought of as "owner" or "project".
+A permissions system then defines policies that apply to those classes.
+
+For example, consider a store shared by multiple teams from the News and Sport production teams of an organisation.
+Each team have the ability to read and write their own content, and no access to the other team's content.
+However in some cases it is necessary to share a particular Source (e.g. to work on a shared story) to the other team.
+
+| Resource       | Classes | Comments |
+| -------------- | ------- | -------- |
+| Source Sport A | sport | |
+| Source Sport B | sport | |
+| Source News X  | news, sport_share | This item is shared across to Sport |
+| Source News Y  | news | |
+
+The permissions policies are then implemented as:
+
+- Users in the "sport" user group have read and write access to all content in the "sport" class
+- Users in the "news" user group have read access to all content in the "news" class
+- Users in the "sport" user group have read access to all content in the "news" class
```

Comment:
Should this be `sport_share`?

Reply:
Yeah, that's a copy-paste error, sorry. Now fixed